### PR TITLE
Added possibility for user to reset his password

### PR DESCRIPTION
**Pull request by IhorF**
Added possibility for user to reset his password

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/service/AccountVerificationByEmailService.java [hunk 1]
```diff
@@ -32,4 +35,11 @@ public void accountVerification(String token) {
         }
         accountRepository.save(editedAcc);
     }
+
+    public void verificationTokenForResetPassword(String token) {
+        String identifier = jwtTokenForResetPassword.decryptToken(token);
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/AccountVerificationByEmailService.java]: Please take into account that there was a comment durring last demo. It is preferable to this method to return boolean instead of trhrowing exception.

**Comment by antonbakalets** [comment]
Please resolve conflicts.

File: src/main/resources/mailTemplates/restorePasswordMailTemplate.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<html>
```
Comment (antonbakalets) [inline src/main/resources/mailTemplates/restorePasswordMailTemplate.txt]: I suggest this file type and other template files type can be changed to html. The IDE will help with formating. But this is out of scope of this task.

Comment (IhorF) [reply src/main/resources/mailTemplates/restorePasswordMailTemplate.txt]: Thanks, I'll change it.
